tests/rados/test_scrub_enhancement.py

```diff
@@ -0,0 +1,204 @@
+            # Retiving the snapshot object
+            sna_obj, _ = acting_osd_node.exec_command(sudo=True, cmd=cmd_get_snaObj)
+            sna_obj = sna_obj.strip()
+            sna_obj_split_result = sna_obj.split(".")
+            sna_key_obj_list = sna_obj.split("\t")
+            corrupt_key = sna_obj_split_result[1].strip()
+            # Creating the corrupted key
+            corrupt_key = corrupt_key + "CORRUPTED"
+            # Getting the correct key
+            correct_key = sna_key_obj_list[1].strip()
+            # Creating mount directory in the acting osd to create the object output
+            cmd_create_dir = f"mkdir -p {mount_point}"
+            acting_osd_node.exec_command(sudo=True, cmd=cmd_create_dir)
+            cmd_str_obj = (
+                f"{cmd_base}mount {mount_point}/ -- ceph-kvstore-tool bluestore-kv  /var/lib/ceph/osd/"
+                f"ceph-{primary_osd} get p  {correct_key} out /mnt/output"
+            )
+            # Removing the snapshot object
+            acting_osd_node.exec_command(sudo=True, cmd=cmd_str_obj)
+            log.info(f"Removing the object in the osd {primary_osd}")
+            cmd_rm_obj = (
+                f"{cmd_base} ceph-kvstore-tool bluestore-kv /var/lib/ceph/osd/ceph-{primary_osd} rm p "
+                f"{correct_key}"
+            )
+            acting_osd_node.exec_command(sudo=True, cmd=cmd_rm_obj)
+            # Checking the snapshot object is removed
+            no_obj_tup = acting_osd_node.exec_command(sudo=True, cmd=cmd_no_obj)
+            assert int(no_obj_tup[0].strip()) == 0
+            log.info("The snapshot object is deleted")
+
+            # inserting the corrupted snapshot object
+            log.info("Creating the corrupted SNA_ entry")
+            cmd_insert_obj = (
+                f"{cmd_base}mount {mount_point}/ -- ceph-kvstore-tool bluestore-kv /var/lib/ceph/osd/"
+                f"ceph-{primary_osd} set p {corrupt_key} in /mnt/output"
+            )
+            acting_osd_node.exec_command(sudo=True, cmd=cmd_insert_obj)
+            time.sleep(5)
+            no_obj_tup = acting_osd_node.exec_command(sudo=True, cmd=cmd_no_obj)
+            assert int(no_obj_tup[0].strip()) == 1
+            log.info("The corrupted snapshot object is inserted")
+
+            if not rados_obj.change_osd_state(action="start", target=primary_osd):
+                log.error(f"Unable to start the OSD : {primary_osd}")
+                raise Exception("Execution error")
+
+            # scrubbing/deep scrubbing the pool
+            scrub_cmd = f"ceph osd pool {test_input}  {pool_name}"
+            old_scrub_time = get_pgid_timestamp(pg_id, test_input)
+            rados_obj.run_ceph_command(scrub_cmd)
+            time_end = time.time() + 60 * 30
+            counter = 0
+            while time.time() < time_end:
+                log.info(f"{test_input} is in progress")
+                if counter == 20:
+                    rados_obj.run_ceph_command(scrub_cmd)
```

Comment:
Any reason to run the scrub command every 500 Seconds?

Reply:
And the duration of the check is only 1800 seconds. I believe the scrub time taken depends on the number of objects in the pool, no of PGs and if there are already PGs in scrub/deep scrub, which are triggered by ceph, it might take longer.

Let us keep BM in mind and increase the timeout.

Also, another thing to keep in mind is the scrub/deep-scrub is not triggered if the PG in question is not active+ clean, meaning if it's recovering or backfilling due to OSD down and start, it will not trigger.

Better to add a check to see if the PG that the scrub is being triggered for, is checked for active+ clean

Reply:
Scrub was not triggered so included the command. but if I am executing the scrub/deep-scrub after the pg active+clean the scrub/deep-scrub is triggered. 
Regarding duration time, I am checking the time with a couple of runs I will increase the time if it fails. Please suggest if is it fine or do I need to increase the time I will increase.